public/js/util/defer.js

```diff
@@ -0,0 +1,16 @@
+"use strict";
+
+function defer() {
+  let resolve, reject;
+  const promise = new Promise(function() {
+    resolve = arguments[0];
+    reject = arguments[1];
+  });
+  return {
+    resolve: resolve,
+    reject: reject,
+    promise: promise
+  };
+}
+
+module.exports = defer;
```

Comment:
added defer to be in line with devtools. It might make sense to move this to the vender directory... Although probably not a big deal